File: rust/src/client/shell/input_reader.rs
use text_io::read;

pub fn read_input() -> String {
    print!("Your Answer >>> ");
    let mut user_input: String = read!("{}\n");
    user_input = user_input.trim().to_lowercase().to_string();
    read_exit(&user_input);
    user_input
}

pub fn read_number() -> f64 {
    loop {
        let user_input = read_input();

        match user_input.parse::<f64>() {
            Ok(value) => return value,
            Err(_) => println!("Invalid number. Please try again."),
        }
    }
}

fn read_exit(user_input: &str) {
    if user_input == "exit" || user_input == "e" {
        println!("Thank you for using the Shapes CLI!");
        std::process::exit(0);
    }
}

File: ruby/lib/shapes/client/shell/input_reader.rb
# frozen_string_literal: true

def read_input
    print "Your Answer >>> "
    user_input = gets.chomp
    user_input = user_input.strip.downcase
    read_exit(user_input)
    user_input
end

def read_number
    loop do
        user_input = read_input
        begin
            return Float(user_input)
        rescue ArgumentError
            puts "Invalid number. Please try again."
        end
    end
end

def read_exit(user_input)
    return unless %w[exit e].include?(user_input)

    puts "Thank you for using the Shapes CLI!"
    exit(0)
end
